lib/db/waitlist.ts

```diff
@@ -0,0 +1,73 @@
+// This is a library with functions to interact with the waitlist table in the database.
+import { Waitlist } from "@prisma/client";
+import prisma from "../prisma";
+
+/**
+ * @param address - The user's address.
+ * @returns all waitlists that the user has created.
+ **/
+export const getUserWaitlists = async (address: string) => {
+  return await prisma.waitlist.findMany({
+    where: {
+      userAddress: address!,
+    },
+    include: {
+      _count: {
+        select: { waitlistedUsers: true },
+      },
+      waitlistRequirements: true,
+    },
+  });
+};
+
+/**
+ * @param address - The user's address.
+ * @returns the number of waitlists the user has created.
+ */
+export const getUserWaitlistsCount = async (
+  address: string
+): Promise<number> => {
+  return (
+    await prisma.waitlist.findMany({
```

Comment:
Here you should use a function called count()

check here: https://www.prisma.io/docs/orm/prisma-client/queries/aggregation-grouping-summarizing#count-records